### PR TITLE
20413: Implemented. Now 0 and 1 treated as boolean if specified.

### PR DESCRIPTION
**Pull request by kelebra**

### CODE REVIEW

**Comment by kelebra** [comment]
Refs #20413

File: akka-http-tests/src/test/scala/akka/http/scaladsl/server/directives/ParameterDirectivesSpec.scala [hunk 1]
```diff
@@ -122,6 +122,11 @@ class ParameterDirectivesSpec extends FreeSpec with GenericRoutingSpec with Insi
         parameter('really.as[Boolean]) { echoComplete }
       } ~> check { responseAs[String] shouldEqual "false" }
     }
+    "extract numeric parameter value as Boolean" in {
+      Get("/?really=1") ~> {
+        parameter('really.as[Boolean]) { echoComplete }
+      } ~> check { responseAs[String] shouldEqual "true" }
+    }
```
Comment (ktoso) [inline akka-http-tests/src/test/scala/akka/http/scaladsl/server/directives/ParameterDirectivesSpec.scala]: Please add a test for the 0 case too

**Comment by kelebra** [comment]
Update: Added test for false scenario

**Comment by johanandren** [comment]
OK TO TEST

**Comment by johanandren** [comment]
LGTM

**Comment by ktoso** [comment]
LGTM, thanks!

**Pull request merged**